crates/factor-key-value/src/lib.rs

```diff
@@ -20,11 +20,19 @@ use store::{store_from_toml_fn, StoreFromToml};
 
 pub use store::MakeKeyValueStore;
 
-#[derive(Default)]
 pub struct KeyValueFactor {
     store_types: HashMap<&'static str, StoreFromToml>,
+    default_store_type: &'static str,
```

Comment:
Just had a chat with @rylev about how to handle this in the sqlite factor; his implementation is very different from this one which I think is fine for now but we should try to converge on a single pattern "pretty soon".

Reply:
I'm obviously biased since I wrote it, but I do like the other implementation for being completely Spin CLI agnostic. A Spin CLI specific approach can be bundled into the crate for convenience, but the main implementation has no specific knowledge of how Spin CLI handles things. 

Unless there's objections, I can open a PR that moves this implementation to mirror that approach so we can discuss whether we think it's the right way to go.

Reply:
@rylev that sounds good to me. I like how your approach is agnostic and doesn't define what "string" is default.